src/contracts/pods/EigenPod.sol

```diff
@@ -164,267 +175,355 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+        uint64 validatorNewBalanceGwei = BeaconChainProofs.getBalanceFromBalanceRoot(validatorIndex, proofs.balanceRoot);        
+
         // verify ETH validator proof
-        bytes32 beaconStateRoot = eigenPodManager.getBeaconChainStateRoot(oracleBlockNumber);
+        bytes32 latestBlockHeaderRoot = eigenPodManager.getBeaconChainStateRoot(oracleTimestamp);
+
+        // verify that the provided state root is verified against the oracle-provided latest block header
+        BeaconChainProofs.verifyStateRootAgainstLatestBlockHeaderRoot(proofs.beaconStateRoot, latestBlockHeaderRoot, proofs.latestBlockHeaderProof);
+
+        // verify validator fields
+        BeaconChainProofs.verifyValidatorFields(
+            validatorIndex,
+            proofs.beaconStateRoot,
+            proofs.validatorFieldsProof,
+            validatorFields
+        );
  
-        /**
-         * If validator's balance is zero, then either they have fully withdrawn or they have been slashed down zero.
-         * If the validator *has* been slashed, then this function can proceed. If they have *not* been slashed, then
-         * the `verifyAndProcessWithdrawal` function should be called instead.
-         */
-        if (validatorCurrentBalanceGwei == 0) {
-            uint64 slashedStatus = Endian.fromLittleEndianUint64(validatorFields[BeaconChainProofs.VALIDATOR_SLASHED_INDEX]);
-            require(slashedStatus == 1, "EigenPod.verifyOvercommittedStake: Validator must be slashed to be overcommitted");
-            //Verify the validator fields, which contain the validator's slashed status
-            BeaconChainProofs.verifyValidatorFields(
-                validatorIndex,
-                beaconStateRoot,
-                proofs.validatorFieldsProof,
-                validatorFields
-            );
-        }
         // verify ETH validator's current balance, which is stored in the `balances` container of the beacon state
-       BeaconChainProofs.verifyValidatorBalance(
+        BeaconChainProofs.verifyValidatorBalance(
             validatorIndex,
-            beaconStateRoot,
+            proofs.beaconStateRoot,
             proofs.validatorBalanceProof,
             proofs.balanceRoot
         );
+        //verify provided slot is valid against beaconStateRoot
+        BeaconChainProofs.verifySlotRoot(
+            proofs.beaconStateRoot,
+            proofs.slotRoot,
+            proofs.slotProof
+        );
+
+        uint64 currentRestakedBalanceGwei = validatorInfo.restakedBalanceGwei;
+
+        // calculate the effective (pessimistic) restaked balance
+        uint64 newRestakedBalanceGwei = _calculateRestakedBalanceGwei(validatorNewBalanceGwei);
+
+        //update the balance
+        validatorInfo.restakedBalanceGwei = newRestakedBalanceGwei;
+
+        //update the most recent balance update slot
+        validatorInfo.mostRecentBalanceUpdateSlot = Endian.fromLittleEndianUint64(proofs.slotRoot);
```

Comment:
not sure if the optimizer will automatically batch these in a single SSTORE -- might be worth it to copy to memory, updated, then copy to storage

Reply:
Not sure what you mean? validatorInfo is in memory

Reply:
ohhh never mind, somehow missed the storage update below.